src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -163,24 +166,30 @@ void Simulator::pack_actuator_message(mavlink_hil_actuator_controls_t &msg, unsi
 	msg.mode = mode_flag_custom;
 	msg.mode |= (armed) ? mode_flag_armed : 0;
 	msg.flags = 0;
+
+	return msg;
 }
 
 void Simulator::send_controls()
 {
-	for (unsigned i = 0; i < (sizeof(_actuator_outputs_sub) / sizeof(_actuator_outputs_sub[0])); i++) {
+	// copy new actuator data if available
+	bool updated = false;
+	orb_check(_actuator_outputs_sub, &updated);
 
-		if (_actuator_outputs_sub[i] < 0 || _actuators[i].timestamp == 0) {
-			continue;
-		}
+	if (updated) {
+		actuator_outputs_s actuators{};
+		orb_copy(ORB_ID(actuator_outputs), _actuator_outputs_sub, &actuators);
+
+		if (actuators.timestamp > 0) {
+			const mavlink_hil_actuator_controls_t hil_act_control = pack_actuator_message(actuators);
```

Comment:
So you removed anything but index 0 without even replacing it with a comment or commit message. I think we should document things like that.

Reply:
There's only a single HIL_ACTUATOR_CONTROLS message with no notion of different groups.

Reply:
Ok fair enough.